cmd/node_problem_detector.go

```diff
@@ -50,7 +81,9 @@ func main() {
 		os.Exit(0)
 	}
 
+	nodeName := mustGetNodeName()
+
 	k := kernelmonitor.NewKernelMonitorOrDie(*kernelMonitorConfigPath)
-	p := problemdetector.NewProblemDetector(k, *apiServerOverride)
+	p := problemdetector.NewProblemDetector(k, *apiServerOverride, nodeName)
```

Comment:
Yeah, I agree.

In fact, following the K8s convention, we should add an `option/` directory under `cmd/` and define the config object and flags there.

Feel free to file an issue or send a PR. :)